packages/tools/devtools/devtools-core/src/BaseDevtools.ts

```diff
@@ -0,0 +1,450 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { IAudience } from "@fluidframework/container-definitions";
+import type { IFluidLoadable } from "@fluidframework/core-interfaces";
+import type { IClient } from "@fluidframework/driver-definitions";
+
+import type { AudienceClientMetadata } from "./AudienceMetadata.js";
+import type { ContainerKey, FluidObjectId, HasContainerKey } from "./CommonInterfaces.js";
+import { ContainerStateChangeKind } from "./Container.js";
+import type { ContainerStateMetadata } from "./ContainerMetadata.js";
+import type { ContainerDevtoolsFeatureFlags } from "./Features.js";
+import type { IContainerDevtools } from "./IContainerDevtools.js";
+import type { AudienceChangeLogEntry, ConnectionStateChangeLogEntry } from "./Logs.js";
+import {
+	DataVisualizerGraph,
+	type FluidObjectNode,
+	type RootHandleNode,
+	defaultVisualizers,
+} from "./data-visualization/index.js";
+import {
+	AudienceSummary,
+	ContainerDevtoolsFeatures,
+	ContainerStateChange,
+	ContainerStateHistory,
+	DataVisualization,
+	GetAudienceSummary,
+	GetContainerDevtoolsFeatures,
+	GetContainerState,
+	GetDataVisualization,
+	GetRootDataVisualizations,
+	type IDevtoolsMessage,
+	type ISourcedDevtoolsMessage,
+	type InboundHandlers,
+	type MessageLoggingOptions,
+	RootDataVisualizations,
+	handleIncomingWindowMessage,
+	postMessagesToWindow,
+} from "./messaging/index.js";
+
+/**
+ * Abstract base class for devtools implementations.
+ * Contains shared functionality between ContainerDevtools and DataObjectDevtools.
+ *
+ * @internal
```

Comment:
We shouldn't need to package export this, so it shouldn't need the `@internal` release tag.